exp/api/v1beta1/azuremanagedmachinepool_webhook.go

```diff
@@ -99,6 +99,10 @@ func (r *AzureManagedMachinePool) ValidateUpdate(oldRaw runtime.Object, client c
 		}
 	}
 
+	// TODO is there a need to do MaxPods validation here?
+	// It's not obvious from kubelet code that there is an enforced "valid range",
+	// But perhaps AKS enforces its own limits
```

Comment:
not sure, @alexeldeib thoughts?

Reply:
The maxPods range actually varies based on the [networkPlugin](https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/26b3a43dcfd794c60bbe3736cddfd6ff09be7d70/exp/api/v1beta1/azuremanagedcontrolplane_types.go#L73) configured by the user. 

https://docs.microsoft.com/fi-fi/azure/aks/configure-azure-cni#configure-maximum---new-clusters

So it's a bit tricky to do this. If we have to validate it, we first need to get the amcp object to check the networkPlugin configured.

Reply:
I wouldn't want to block progress on solving this (based on (1) the precedent of the existing rather absent input validations for this cluster type and (2) the experimental nature of the AzureManagedCluster surface area).

But should we keep the // TODO comment as a reminder to revisit it later when it's appropriate to reinforce this stuff?

Reply:
I think we should validate for a range of 10-250, the aks api validates for the same. 
I tried this for the both kubenet and azure and returns the same :) 
```
lochanrn@Lochans-MacBook-Pro ~ % az aks create -g rnl-rg -n aks-test --max-pods 300 --network-plugin kubenet
(MaxPodsLimitExceeded) The AgentPoolProfile 'nodepool1' has an invalid maxPods, the maxPods(300) should be smaller than or equal to 250. Please refer to aka.ms/aks-min-max-pod for more details.
Code: MaxPodsLimitExceeded
Message: The AgentPoolProfile 'nodepool1' has an invalid maxPods, the maxPods(300) should be smaller than or equal to 250. Please refer to aka.ms/aks-min-max-pod for more details.
Target: agentPoolProfile.kubernetesConfig.kubeletConfig.maxPods
```

During the update we need to ensure it is [immutable](https://docs.microsoft.com/fi-fi/azure/aks/configure-azure-cni#configure-maximum---existing-clusters)

Reply:
I basically agree with @LochanRn's assessment, but I agree it's not a blocker given the existing validation. I do think we should go for the cross object validation eventually, or else have a simple validation on < 250 (the higher of the two limits) for now

Reply:
Thanks for the feedback, I've added validations that match the current AKS requirements.